tests/publiccloud/ipa.pm

```diff
@@ -159,6 +138,80 @@ sub post_fail_hook {
     assert_script_run("aws ec2 delete-key-pair --key-name " . $self->{'ipa_ssh_key_name'});
 }
 
+sub az_run_ipa {
+    my ($self) = @_;
+
+    my $clientid      = get_required_var('PUBLIC_CLOUD_KEY_ID');
+    my $clientsecret  = get_required_var('PUBLIC_CLOUD_KEY_SECRET');
+    my $subscription  = get_required_var('PUBLIC_CLOUD_SUBSCRIPTION_ID');
+    my $tenantid      = get_required_var('PUBLIC_CLOUD_TENANT_ID');
+    my $image         = get_required_var('PUBLIC_CLOUD_IMAGE_ID');
+    my $instance_type = get_var('PUBLIC_CLOUD_INSTANCE_TYPE', 'Standard_A2');
+    my $region        = get_var('PUBLIC_CLOUD_REGION', 'westeurope');
+
+    my $credentials = <<EOT;
+{
+  "clientId": "$clientid", 
+  "clientSecret": "$clientsecret", 
+  "subscriptionId": "$subscription", 
+  "tenantId": "$tenantid", 
+  "activeDirectoryEndpointUrl": "https://login.microsoftonline.com", 
+  "resourceManagerEndpointUrl": "https://management.azure.com/", 
+  "activeDirectoryGraphResourceId": "https://graph.windows.net/", 
+  "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/", 
+  "galleryEndpointUrl": "https://gallery.azure.com/", 
+  "managementEndpointUrl": "https://management.core.windows.net/"
+}
+EOT
+
+    save_tmp_file("azure_credentials.txt", $credentials);
+    assert_script_run('curl -O ' . autoinst_url . "/files/azure_credentials.txt");
+
+    assert_script_run('ssh-keygen -q -N "" -f ~/.ssh/id_rsa');
```

Comment:
I've seen in the repo this way of creating the ssh-pair

    ssh-keygen -b 2048 -t rsa -q -N "" -f ~/.ssh/id_rsa

It will probably produce the same results though

Reply:
True it will properly produce the same output, but I like the more specific version of you better. Will change it.